src/sotoki/utils/preparation.py

```diff
@@ -16,11 +16,31 @@
 
 from .shared import logger
 from .misc import has_binary, get_available_memory
-from ..constants import UTF8
+from ..constants import UTF8, UTF16LE
 
 has_gnusort = has_binary("sort")
 
 
+def reencode_file(src: pathlib.Path):
+    """Reencode a file from dump format (UTF-16-LE as of March 2024) to expected format (UTF8)
+
+    This is based on a streaming on-the-fly reencoding of file chunks to limit memory pressure.
+
+    During few instants, there will be two versions of the same content on the filesystem, one
```

Comment:
I don't think “during few instants” is a wise addition. The same-file-twice duration is irrelevant but the comment makes it looks like the whole process is short. It's entirely dependent on the host's performance (disk I/O mostly) and given some of those files are tens of GB, just witting them on disk takes a long time.

Also, it's good to inform that both versions will be present on disk at the same time but it shouldn't affect disk provisioning as this is happening in preparation step. A sotoki run always requires twice the dump size of disk space as content is to be copied into ZIM (it's more complicated as you imagine but that's the rule of thumb we give to users)